accessible/workspace_treeview.component.js

```diff
@@ -0,0 +1,225 @@
+                  [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap['markBelowButton'], 'blockly-button', noNextConnectionHTMLText(block))"  
+                  [attr.aria-level]="level+2" aria-selected=false>
+                <button [id]="idMap['markBelowButton']" (click)="clipboardService.markConnection(block.nextConnection)" 
+                    [disabled]="noNextConnectionHTMLText(block)">mark spot below</button>
+              </li>
+              <li [id]="idMap['markAbove']" role="treeitem" 
+                  [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap['markAboveButton'], 'blockly-button', noPreviousConnectionHTMLText(block))" 
+                  [attr.aria-level]="level+2" aria-selected=false>
+                <button [id]="idMap['markAboveButton']" (click)="clipboardService.markConnection(block.previousConnection)" 
+                    [disabled]="noPreviousConnectionHTMLText(block)">mark spot above</button>
+              </li>
+              <li [id]="idMap['sendToSelectedListItem']" role="treeitem" 
+                  [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap['sendToSelectedButton'], 'blockly-button', utilsService.getMarkedBlockCompatibilityHTMLText(clipboardService.isBlockCompatibleWithMarkedConnection(block)))" 
+                  [attr.aria-level]="level+2" aria-selected=false>
+                <button [id]="idMap['sendToSelectedButton']" (click)="sendToSelected(block)" 
+                    [disabled]="utilsService.getMarkedBlockCompatibilityHTMLText(clipboardService.isBlockCompatibleWithMarkedConnection(block))">move to marked spot</button>
+              </li>
+              <li [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap['deleteButton'], 'blockly-button')"  [id]="idMap['delete']" role="treeitem" aria-selected=false [attr.aria-level]="level+2">
+                <button [id]="idMap['deleteButton']" (click)="block.dispose(true)">delete</button>
+              </li>
+            </ol>
+          </li>
+          <div *ngFor="#inputBlock of block.inputList; #i = index">
+            <field-view *ngFor="#field of getInfo(inputBlock)" [field]="field"></field-view>
+            <tree-view *ngIf="inputBlock.connection && inputBlock.connection.targetBlock()" [block]="inputBlock.connection.targetBlock()" [isTopBlock]="false" [level]="level"></tree-view>
+            <li #inputList [attr.aria-level]="level + 1" [id]="idMap['inputList' + i]" 
+                [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr('blockly-menu', idMap['inputMenuLabel' + i])" 
+                *ngIf="inputBlock.connection && !inputBlock.connection.targetBlock()" (keydown)="treeService.onKeypress($event, tree)">
+              <label [id]="idMap['inputMenuLabel' + i]"> {{utilsService.getInputTypeLabel(inputBlock.connection)}} {{utilsService.getBlockTypeLabel(inputBlock)}} needed: </label>
```

Comment:
@NeilFraser how do I go about i18n-ing this line? Do I need to worry about making a Block.msg.X for each input type or is that already baked in somewhere?